streams/jvm/src/main/scala/scalaz/zio/stream/platform.scala

```diff
@@ -0,0 +1,51 @@
+package scalaz.zio.stream
+
+import java.io.{ InputStream, OutputStream }
+
+import scalaz.zio._
+import scalaz.zio.blocking._
+
+trait ZStreamPlatformSpecific {
+
+  /**
+   * Uses the provided `TaskR` value to create a [[ZStream]] of byte chunks, backed by
+   * the resulting `InputStream`. When data from the `InputStream` is exhausted,
+   * the stream will close it.
+   */
+  def fromInputStream[R](
+    is: TaskR[R, InputStream],
+    chunkSize: Int = 4096
+  ): ZStreamChunk[R with Blocking, Throwable, Byte] =
+    ZStreamChunk {
+      ZStream.managed(is.toManaged(is => effectBlocking(is.close()).orDie)) { is =>
+        effectBlocking {
```

Comment:
`effectBlocking(is.close()).option.unit`, maybe? I am not sure if it's better to die on close of the stream or just ignore the error and continue.

Reply:
Hmm not sure either. But dying here can be circumvented by the user by sandboxing; swallowing the error cannot.